### PR TITLE
Detect process-id on Java 8 via ManagementFactory instead of JNR/POSIX

### PR DESCRIPTION
**Pull request by mcculls**
# What Does This Do
Relies on an assumption that the initial part of the vmId from `ManagementFactory` is our process id.

This is true for all JVMs based on OpenJDK, but may not be true for other JVMs that don't follow this convention:
https://github.com/openjdk/jdk/blob/jdk8-b120/jdk/src/share/classes/sun/management/VMManagementImpl.java#L145

Note that JVMs after Java8 will continue to use the standard `ProcessHandle.current().pid()` API

# Motivation
PR #4397 added the process-id to all root spans, which means that the process-id is now requested as part of building the tracer in `premain`. Unfortunately the JNR/POSIX layer which is used for Java8 only takes some time to initialize, and this leads to a noticeable increase in startup time.

On my M1 mac the JNR/POSIX approach takes 0.5 seconds, whereas using `ManagementFactory` takes 0.03 seconds.

### CODE REVIEW

File: internal-api/src/main/java/datadog/trace/util/PidHelper.java [hunk 1]
```diff
@@ -40,7 +41,16 @@ private static String findPid() {
       }
     }
     if (pid.isEmpty()) {
-      pid = Strings.trim(Fallback.supplier.get());
+      try {
+        // assumption: first part of runtime vmId is our process id
```
Comment (PerfectSlayer) [inline internal-api/src/main/java/datadog/trace/util/PidHelper.java:45]: Is it useful to add the link to the OpenJDK you have in your PR description? (or git blame will be enough to find back the explaination?)

**Review by PerfectSlayer — APPROVED** [review]
LGTM

File: internal-api/src/main/java/datadog/trace/util/PidHelper.java (hunk 1 above)
Comment (mcculls) [reply internal-api/src/main/java/datadog/trace/util/PidHelper.java:45]: I think the commit history + the comment is enough - the key part is this is an assumption, not a guarantee

**Review by richardstartin — APPROVED** [review]
🚀

**Pull request merged**